### ISSUE TITLE
Can keras model be trained without label?

### ISSUE THREAD

**Issue by Epiphqny** [keras-team/keras#7882]
Suppose I want to train a model to optimize a loss, but without a limited target, how can i train the model?

**Comment by ericwu09** [comment]
You still need an output, but in your loss function you can ignore the y_true. Then generate dummy values for the y values in the training call.

**Comment by Epiphqny** [comment]
Thanks, that's a good idea！

**Comment by bhalajin** [comment]
@ericwu09 or @Epiphqny , can you provide a sample code for the suggestion made?

**Comment by Epiphqny** [comment]
@bhalajin just write your own loss function which does not use y_true, and set any value to y_true will make no difference

**Comment by ericwu09** [comment]
For instance:
`
def dummy_loss(y_true, y_pred):
    return y_pred
`

**Comment by nwoodruff149** [comment]
Tell me if I've got this wrong, but wouldn't it be a better idea to return y_true rather than y_pred? Surely using the predicted value as its own loss would just cause it to minimize its predictions, and by returning y_true couldn't you input the loss yourself directly when fitting?

**Comment by ybsave** [comment]
@nwoodruff149 Only y_pred is based on your network; y_true has no relation with your network. Therefore, minimizing y_true is meaningless, as it is constant.

**Comment by h227-chen** [comment]
Silly question: how does my loss function know which is y_true or y_pred? Or in other words, how y_true and y_pred pass in? 
Are they key words in Keras? Or are their positions fixed?
